staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/applyconfiguration.go

```diff
@@ -291,16 +317,40 @@ func (g *applyConfigurationGenerator) generateMemberWithForSlice(sw *generator.S
 	sw.Do("}\n", memberParams)
 }
 
-func (g *applyConfigurationGenerator) generateMemberWithForMap(sw *generator.SnippetWriter, memberParams memberParams) {
-	sw.Do("// With$.Member.Name$ puts the entries into the $.Member.Name$ field in the declarative configuration\n", memberParams)
+func (g *applyConfigurationGenerator) generateMemberWithForAtomicSlice(sw *generator.SnippetWriter, member types.Member, memberParams memberParams) {
+	sw.Do("// WithNew$.Member.Name$ replaces the $.Member.Name$ field in the declarative configuration with given values\n", memberParams)
 	sw.Do("// and returns the receiver, so that objects can be build by chaining \"With\" function invocations.\n", memberParams)
-	sw.Do("// If called multiple times, the entries provided by each call will be put on the $.Member.Name$ field,\n", memberParams)
-	sw.Do("// overwriting an existing map entries in $.Member.Name$ field with the same key.\n", memberParams)
-	sw.Do("func (b *$.ApplyConfig.ApplyConfiguration|public$) With$.Member.Name$(entries $.MemberType|raw$) *$.ApplyConfig.ApplyConfiguration|public$ {\n", memberParams)
+	sw.Do("// If called multiple times, the $.Member.Name$ field is set to the values of the last call.\n", memberParams)
```

Comment:
Can we mark the non-new method deprecated with a `// Deprecated: ...`?

Reply:
Done